metrics/sockets.go

```diff
@@ -0,0 +1,33 @@
+package metrics
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+// getSocketCount returns the number of open sockets
+func getSocketCount() uint64 {
+	var socketCount uint64
+	pid := os.Getpid()
+	base := fmt.Sprintf("/proc/%d/fd", pid)
+	fds, err := ioutil.ReadDir(base)
+	if err != nil {
+		// Should we log something here?
+		return 0
+	}
+
+	for _, fd := range fds {
+		sl, err := os.Readlink(fmt.Sprintf("%s/%s", base, fd.Name()))
```

Comment:
there are other ways to get tcp connection information from `/proc` so I'm not sure if we might want to use them. maybe that's a follow-up item. This seems good for now